wagtail/admin/views/collections.py

```diff
@@ -57,6 +70,37 @@ class Edit(EditView):
     context_object_name = 'collection'
     header_icon = 'folder-open-1'
 
+    def get_queryset(self):
+        if self.request.user.is_superuser:
+            # Only return descendants of the root node, so that the root is not editable
+            return Collection.get_first_root_node().get_descendants()
+        else:
+            return self.permission_policy.collections_user_has_permission_for(
+                self.request.user, 'change'
+            )
+
+    def get_form(self, form_class=None):
+        form = super().get_form(form_class)
+        user = self.request.user
+        # If this instance is a collection used to assign permissions for this user,
+        # do not let the user move this collection.
+        if self.permission_policy.user_has_any_permission_directly_on_instance(
+            user, ['add', 'change', 'delete'], form.instance
+        ):
+            form.fields['parent'].widget = HiddenInput()
```

Comment:
I tried this but if I disable the parent form field, then it does not get submitted at all - which means I would have to make the parent field optional in the form. That is how the CollectionForm was originally written but then we needed to fill in the parent (see the diff for wagtail/admin/forms/collections.py in this PR). So instead of adding disabled = True, I added code in the `form_valid` method to prevent someone from posting a parent id they should not be able to send from the front end form. My test cases cover the hacks I could think of.

If you would like me to go back to the original form __init__ logic plus disabled = True to handle this, let me know and I will rearrange.

Reply:
Okay, so, in these cases, maybe you can just remove the field completely? That way, it isn't going to display, won't factor into validation, and the instance will just keep it's current value.

If you couple that with `self.instance.parent = value` for new objects, so that they have a value, I think all should be well.

Reply:
Disabling a form field after the field is initialized does disable it in the rendered html. Unfortunately it doesn't tell the form class that the field is disabled so the logic that always keeps the initial value doesn't work. I solved this by putting the logic I had originally put in the create and edit views into the CollectionForm where it should have been in the first place. 

The one slightly unusual aspect of this is that because the logic depends on the current user and current permission policy, I had to pass both of those items from the view into the form itself.

Reply:
@cnk did you see my suggestion about simply removing the field? Forms only update model fields which they themselves have a field for, so juts popping it within `__init__()` would totally work. 

With this new approach, I feel like the distinction between view/form has become a bit blurry. I expect much the same could be achieved by keeping the user-specific logic in the view, and passing more refined values to the form, e.g a queryset for the parent field.

Reply:
I guess I do not understand what you mean by "removing the field". Do you mean something like `form.fields['parent'].pop()` in the view's `get_form` method? Does that work? I'll fall back a commit and try that approach. 

From the reading I did, pushing all the logic into the form object is the preferred django approach - my only concern was that it involved changing the signature of `CollectionForm.__init__`

Reply:
So removing the field from the form does work and gives the desired behavior. This is ready for review (assuming the automated checks all pass).  (Thank goodness we use 'squash' when merging PRs. I would hate for the history of this PR to end up the the project history)

Reply:
@cnk I feel like you took my idea and made it even better! This latest iteration feels a lot clearer.  Are you happier with it?

Reply:
I am shocked that removing the field works but am pleased with the result. It took care of all the problems I was having when just disabling the field. I am satisfied with this iteration. Ready for review.